services/repository/branch.go

```diff
@@ -65,44 +67,39 @@ func GetBranches(repo *models.Repository, skip, limit int) ([]*git.Branch, int,
 }
 
 // checkBranchName validates branch name with existing repository branches
-func checkBranchName(repo *models.Repository, name string) error {
-	gitRepo, err := git.OpenRepository(repo.RepoPath())
-	if err != nil {
-		return err
-	}
-	defer gitRepo.Close()
-
-	branches, _, err := GetBranches(repo, 0, 0)
-	if err != nil {
-		return err
-	}
-
-	for _, branch := range branches {
-		if branch.Name == name {
+func checkBranchName(ctx context.Context, repo *models.Repository, name string) error {
+	_, err := git.WalkReferences(ctx, repo.RepoPath(), func(refName string) error {
+		branchRefName := strings.TrimPrefix(refName, git.BranchPrefix)
```

Comment:
Should we check whether the `refName` really has the prefix `refs/heads/` ?

Reply:
We have to check both otherwise the tests fail.

We've accepted master and refs/heads/master in this function for a long time and changing it here would be breaking.

Reply:
Yep, I mean should we skip if refName doesn't start with git.BranchPrefix(`refs/heads/`)?

If some refNames don't start with BranchPrefix, the TrimPrefix is a no-op here and the code below will be executed, it seems an unexpected behavior.

Reply:
Oh, sorry, I will check `WalkReferences` again to see if it returns master or something else.